statsd/uds.go

```diff
@@ -107,15 +76,26 @@ func (w *udsWriter) Write(data []byte) (int, error) {
 	if stream {
 		bs := []byte{0, 0, 0, 0}
 		binary.LittleEndian.PutUint32(bs, uint32(len(data)))
-		_, err = w.writeFull(bs, true, true)
+		_, err = w.conn.Write(bs)
```

Comment:
Would it make sense to check number of bytes written? AFAIK Go documentation makes no statement that it the write will be complete if the error is nil.

Reply:
https://cs.opensource.google/go/go/+/refs/tags/go1.21.6:src/io/io.go;l=99

```
// Write writes len(p) bytes from p to the underlying data stream.
// It returns the number of bytes written from p (0 <= n <= len(p))
// and any error encountered that caused the write to stop early.
// Write must return a non-nil error if it returns n < len(p).
// Write must not modify the slice data, even temporarily.
```

So I think it's fine ?

Reply:
Yes should be fine then. Good find.